FirebaseDynamicLinks/Sources/FIRDynamicLinkNetworking.m

```diff
@@ -116,33 +117,66 @@ - (void)resolveShortLink:(NSURL *)url
     @"sdk_version" : FDLSDKVersion
   };
 
-  FIRNetworkRequestCompletionHandler resolveLinkCallback = ^(NSData *data, NSError *error) {
-    NSURL *resolvedURL;
-
-    if (!error && data) {
-      NSDictionary *result = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
-      if ([result isKindOfClass:[NSDictionary class]]) {
-        id invitationIDObject = [result objectForKey:@"invitationId"];
-
-        NSString *invitationIDString;
-        if ([invitationIDObject isKindOfClass:[NSDictionary class]]) {
-          NSDictionary *invitationIDDictionary = invitationIDObject;
-          invitationIDString = invitationIDDictionary[@"id"];
-        } else if ([invitationIDObject isKindOfClass:[NSString class]]) {
-          invitationIDString = invitationIDObject;
+  FIRNetworkRequestCompletionHandler resolveLinkCallback = ^(NSData *data, NSURLResponse *response,
+                                                             NSError *error) {
+    NSURL *resolvedURL = nil;
+    NSHTTPURLResponse *resp = nil;
+
+    if (![response isKindOfClass:[NSHTTPURLResponse class]]) {
+      NSError *err = [NSError
+          errorWithDomain:kGenericErrorDomain
+                     code:0
+                 userInfo:@{@"message" : @"Response should be of type NSHTTPURLResponse."}];
+      handler(resolvedURL, error ? error : err);
+    } else {
+      resp = (NSHTTPURLResponse *)response;
```

Comment:
If I'm not mistaken, `resp` is only used within this `else`'s scope, correct? Perhaps its declaration could be moved  to within the `else` block.

Reply:
Moved.